Description: Add quilt patch to fix mktables pack-J failure in miniperl
 Integrates the mktables pack 'J' fix into the debian/patches quilt
 system so dpkg-source --before-build applies it cleanly and
 dpkg-source -b produces a valid source package. The inner patch
 replaces all pack 'J', refaddr calls in lib/unicore/mktables with
 plain refaddr calls; Perl auto-stringifies the integer for hash
 subscripts, making this semantically identical.
Author: Harkirat S Virk <harkiratsvirk@gmail.com>
Origin: local
Forwarded: no
Last-Update: 2026-05-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- /dev/null
+++ b/debian/patches/local/mktables-pack-J.diff
@@ -0,0 +1,843 @@
+--- a/lib/unicore/mktables
++++ b/lib/unicore/mktables
+@@ -1495,7 +1495,7 @@
+     # every call, and the program is structured so that this is never called
+     # for a non-blessed object.
+ 
+-    return pack 'J', refaddr $addr;
++    return refaddr $addr;
+ }
+ 
+ # These are used only if $annotate is true.
+@@ -1856,7 +1856,7 @@
+             # Use typeglob to give the anonymous subroutine the name we want
+             *$destroy_name = sub {
+                 my $self = shift;
+-                my $addr = pack 'J', refaddr $self;
++                my $addr = refaddr $self;
+ 
+                 $self->$destroy_callback if $destroy_callback;
+                 foreach my $field (keys %{$package_fields{$package}}) {
+@@ -1952,7 +1952,7 @@
+                 # determine using 'eq' for scalars and '==' otherwise.
+                 *$subname = sub ($self, $value) {
+                     use strict "refs";
+-                    my $addr = pack 'J', refaddr $self;
++                    my $addr = refaddr $self;
+                     if (ref $value) {
+                         return if grep { $value == $_ } @{$field->{$addr}};
+                     }
+@@ -1984,7 +1984,7 @@
+                     no strict "refs";
+                     *$subname = sub ($_addr) {
+                         use strict "refs";
+-                        my $addr = pack 'J', refaddr $_addr;
++                        my $addr = refaddr $_addr;
+                         if (ref $field->{$addr} ne 'ARRAY') {
+                             my $type = ref $field->{$addr};
+                             $type = 'scalar' unless $type;
+@@ -2005,7 +2005,7 @@
+                     no strict "refs";
+                     *$subname = sub ($addr) {
+                         use strict "refs";
+-                        return $field->{pack 'J', refaddr $addr};
++                        return $field->{refaddr $addr};
+                     }
+                 }
+             }
+@@ -2015,7 +2015,7 @@
+                 *$subname = sub ($self, $value) {
+                     use strict "refs";
+                     # $self is $_[0]; $value is $_[1]
+-                    $field->{pack 'J', refaddr $self} = $value;
++                    $field->{refaddr $self} = $value;
+                     return;
+                 }
+             }
+@@ -2320,7 +2320,7 @@
+         my $class = shift;
+ 
+         my $self = bless \do{ my $anonymous_scalar }, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Set defaults
+         $handler{$addr} = \&main::process_generic_property_file;
+@@ -2648,7 +2648,7 @@
+         # flag to make sure extracted files are processed early
+         state $seen_non_extracted = 0;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $file = $file{$addr};
+ 
+@@ -2895,7 +2895,7 @@
+         # been added via insert_lines() will be returned in $_ before the file
+         # is read again.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Here the file is open (or if the handle is not a ref, is an open
+         # 'virtual' file).  Get the next line; any inserted lines get priority
+@@ -3076,7 +3076,7 @@
+         # insertion code will sort and coalesce the individual code points
+         # into appropriate ranges.)
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         while (1) {
+ 
+@@ -3151,7 +3151,7 @@
+ #        # an each_line_handler() on the line.
+ #
+ #        my $self = shift;
+-#        my $addr = pack 'J', refaddr $self;
++#        my $addr = refaddr $self;
+ #
+ #        foreach my $inserted_ref (@{$added_lines{$addr}}) {
+ #            my ($adjusted, $line) = @{$inserted_ref};
+@@ -3190,7 +3190,7 @@
+         # Each inserted line is an array, with the first element being 0 to
+         # indicate that this line hasn't been adjusted, and needs to be
+         # processed.
+-        push @{$added_lines{pack 'J', refaddr $self}}, map { [ 0, $_ ] } @lines;
++        push @{$added_lines{refaddr $self}}, map { [ 0, $_ ] } @lines;
+         return;
+     }
+ 
+@@ -3212,7 +3212,7 @@
+ 
+         # Each inserted line is an array, with the first element being 1 to
+         # indicate that this line has been adjusted
+-        push @{$added_lines{pack 'J', refaddr $self}}, map { [ 1, $_ ] } @lines;
++        push @{$added_lines{refaddr $self}}, map { [ 1, $_ ] } @lines;
+         return;
+     }
+ 
+@@ -3222,7 +3222,7 @@
+         # element, and the property in the 2nd.  However, since these lines
+         # can be stacked up, the return is an array of all these arrays.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # If not accepting a list return, just return the first one.
+         return shift @{$missings{$addr}} unless wantarray;
+@@ -3294,7 +3294,7 @@
+         # Hangul syllables in this release only are something else, so if
+         # using such data, we have to override it
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $object = main::property_ref($property{$addr});
+         $object->add_map($FIRST_REMOVED_HANGUL_SYLLABLE,
+@@ -3306,7 +3306,7 @@
+     sub _insert_property_into_line($self) {
+         # Add a property field to $_, if this file requires it.
+ 
+-        my $property = $property{pack 'J', refaddr $self};
++        my $property = $property{refaddr $self};
+         $_ =~ s/(;|$)/; $property$1/;
+         return;
+     }
+@@ -3318,7 +3318,7 @@
+         # only outputs the first instance of each message, incrementing a
+         # count so the totals can be output at the end of the file.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $message = 'Unexpected line' unless $message;
+ 
+@@ -3384,7 +3384,7 @@
+         my $class = shift;
+ 
+         my $self = bless \do{my $anonymous_scalar}, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         while (@_ > 1) {
+             my $default = shift;
+@@ -3400,7 +3400,7 @@
+     sub get_next_defaults($self) {
+         # Iterates and returns the next class of defaults.
+ 
+-        return each %{$class_defaults{pack 'J', refaddr $self}};
++        return each %{$class_defaults{refaddr $self}};
+     }
+ }
+ 
+@@ -3447,7 +3447,7 @@
+         my $class = shift;
+ 
+         my $self = bless \do { my $anonymous_scalar }, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $name{$addr} = shift;
+         $loose_match{$addr} = shift;
+@@ -3507,7 +3507,7 @@
+ 
+     sub new($class, $_addr, $_end, @_args) {
+         my $self = bless \do { my $anonymous_scalar }, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $start{$addr} = $_addr;
+         $end{$addr}   = $_end;
+@@ -3533,7 +3533,7 @@
+     ;
+ 
+     sub _operator_stringify($self, $other="", $reversed=0) {
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Output it like '0041..0065 (value)'
+         my $return = sprintf("%04X", $start{$addr})
+@@ -3556,7 +3556,7 @@
+         # of writing there are 368676 non-special objects, but the standard
+         # form is only requested for 22047 of them - ie about 6%.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return $standard_form{$addr} if defined $standard_form{$addr};
+ 
+@@ -3568,7 +3568,7 @@
+     sub dump($self, $indent) {
+         # Human, not machine readable.  For machine readable, comment out this
+         # entire routine and let the standard one take effect.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $return = $indent
+                     . sprintf("%04X", $start{$addr})
+@@ -3654,7 +3654,7 @@
+         return _union($class, $initialize, %args) if defined $initialize;
+ 
+         $self = bless \do { my $anonymous_scalar }, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Optional parent object, only for debug info.
+         $owner_name_of{$addr} = delete $args{'Owner'};
+@@ -3684,7 +3684,7 @@
+     ;
+ 
+     sub _operator_stringify($self, $other="", $reversed=0) {
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return "Range_List attached to '$owner_name_of{$addr}'"
+                                                 if $owner_name_of{$addr};
+@@ -3749,7 +3749,7 @@
+             if (! defined $arg) {
+                 my $message = "";
+                 if (defined $self) {
+-                    $message .= $owner_name_of{pack 'J', refaddr $self};
++                    $message .= $owner_name_of{refaddr $self};
+                 }
+                 Carp::my_carp_bug($message . "Undefined argument to _union.  No union done.");
+                 return;
+@@ -3774,7 +3774,7 @@
+             else {
+                 my $message = "";
+                 if (defined $self) {
+-                    $message .= $owner_name_of{pack 'J', refaddr $self};
++                    $message .= $owner_name_of{refaddr $self};
+                 }
+                 Carp::my_carp_bug($message . "Cannot take the union of a $type.  No union done.");
+                 return;
+@@ -3821,7 +3821,7 @@
+     }
+ 
+     sub range_count($self) {        # Return the number of ranges in the range list
+-        return scalar @{$ranges{pack 'J', refaddr $self}};
++        return scalar @{$ranges{refaddr $self}};
+     }
+ 
+     sub min($self) {
+@@ -3831,7 +3831,7 @@
+         # and having to worry about changing it as ranges are added and
+         # deleted.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # If the range list is empty, return a large value that isn't adjacent
+         # to any that could be in the range list, for simpler tests
+@@ -3852,7 +3852,7 @@
+         #   range[$i-1]->end < $codepoint <= range[$i]->end
+         # So is in the table if and only iff it is at least the start position
+         # of range $i.
+-        return 0 if $ranges{pack 'J', refaddr $self}->[$i]->start > $codepoint;
++        return 0 if $ranges{refaddr $self}->[$i]->start > $codepoint;
+         return $i + 1;
+     }
+ 
+@@ -3862,7 +3862,7 @@
+         return unless $i;
+ 
+         # contains() returns 1 beyond where we should look
+-        return $ranges{pack 'J', refaddr $self}->[$i-1];
++        return $ranges{refaddr $self}->[$i-1];
+     }
+ 
+     sub value_of($self, $codepoint) {
+@@ -3888,7 +3888,7 @@
+         #   range[$i-1]->end < $codepoint <= range[$i]->end
+         # Returns undef if no such $i is possible (e.g. at end of table), or
+         # if there is an error.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return if $code_point > $max{$addr};
+         my $r = $ranges{$addr};                # The current list of ranges
+@@ -4076,7 +4076,7 @@
+ 
+         Carp::carp_extra_args(\%args) if main::DEBUG && %args;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         if ($operation ne '+' && $operation ne '-') {
+             Carp::my_carp_bug("$owner_name_of{$addr}First parameter to _add_delete must be '+' or '-'.  No action taken.");
+@@ -4721,14 +4721,14 @@
+     }
+ 
+     sub reset_each_range($self) {  # reset the iterator for each_range();
+-        undef $each_range_iterator{pack 'J', refaddr $self};
++        undef $each_range_iterator{refaddr $self};
+         return;
+     }
+ 
+     sub each_range($self) {
+         # Iterate over each range in a range list.  Results are undefined if
+         # the range list is changed during the iteration.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return if $self->is_empty;
+ 
+@@ -4742,7 +4742,7 @@
+     }
+ 
+     sub count($self) {        # Returns count of code points in range list
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $count = 0;
+         foreach my $range (@{$ranges{$addr}}) {
+@@ -4756,14 +4756,14 @@
+     }
+ 
+     sub is_empty($self) { # Returns boolean as to if a range list is empty
+-        return scalar @{$ranges{pack 'J', refaddr $self}} == 0;
++        return scalar @{$ranges{refaddr $self}} == 0;
+     }
+ 
+     sub hash($self) {
+         # Quickly returns a scalar suitable for separating tables into
+         # buckets, i.e. it is a hash function of the contents of a table, so
+         # there are relatively few conflicts.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # These are quickly computable.  Return looks like 'min..max;count'
+         return $self->min . "..$max{$addr};" . scalar @{$ranges{$addr}};
+@@ -5312,7 +5312,7 @@
+         my $class = shift;
+ 
+         my $self = bless \do { my $anonymous_scalar }, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my %args = @_;
+ 
+@@ -5497,7 +5497,7 @@
+     sub ranges {
+         # Returns the array of ranges associated with this table.
+ 
+-        return $range_list{pack 'J', refaddr shift}->ranges;
++        return $range_list{refaddr shift}->ranges;
+     }
+ 
+     sub add_alias {
+@@ -5538,7 +5538,7 @@
+         # release
+         $name = ucfirst($name) unless $name =~ /^k[A-Z]/;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Figure out if should be loosely matched if not already specified.
+         if (! defined $loose_match) {
+@@ -5599,7 +5599,7 @@
+ 
+         # This name may be shorter than any existing ones, so clear the cache
+         # of the shortest, so will have to be recalculated.
+-        undef $short_name{pack 'J', refaddr $self};
++        undef $short_name{refaddr $self};
+         return;
+     }
+ 
+@@ -5618,7 +5618,7 @@
+         # Any name with alphabetics is preferred over an all numeric one, even
+         # if longer.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # For efficiency, don't recalculate, but this means that adding new
+         # aliases could change what the shortest is, so the code that does
+@@ -5721,13 +5721,13 @@
+     }
+ 
+     sub add_description($self, $description) { # Adds the parameter as a short description.
+-        push @{$description{pack 'J', refaddr $self}}, $description;
++        push @{$description{refaddr $self}}, $description;
+ 
+         return;
+     }
+ 
+     sub add_note($self, $note) { # Adds the parameter as a short note.
+-        push @{$note{pack 'J', refaddr $self}}, $note;
++        push @{$note{refaddr $self}}, $note;
+ 
+         return;
+     }
+@@ -5738,7 +5738,7 @@
+ 
+         chomp $comment;
+ 
+-        push @{$comment{pack 'J', refaddr $self}}, $comment;
++        push @{$comment{refaddr $self}}, $comment;
+ 
+         return;
+     }
+@@ -5748,7 +5748,7 @@
+         # context, returns the array of comments.  In scalar, returns a string
+         # of each element joined together with a period ending each.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+         my @list = @{$comment{$addr}};
+         return @list if wantarray;
+         my $return = "";
+@@ -5765,7 +5765,7 @@
+         # Initialize the table with the argument which is any valid
+         # initialization for range lists.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Replace the current range list with a new one of the same exact
+         # type.
+@@ -5819,7 +5819,7 @@
+         # a range equals this one, don't write
+         # the range
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+         my $write_as_invlist = $write_as_invlist{$addr};
+ 
+         # Start with the header
+@@ -6475,7 +6475,7 @@
+     sub set_status($self, $status, $info) {    # Set the table's status
+         # status The status enum value
+         # info Any message associated with it.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $status{$addr} = $status;
+         $status_info{$addr} = $info;
+@@ -6483,7 +6483,7 @@
+     }
+ 
+     sub set_fate($self, $fate, $reason=undef) {  # Set the fate of a table
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return if $fate{$addr} == $fate;    # If no-op
+ 
+@@ -6517,7 +6517,7 @@
+         # Don't allow changes to the table from now on.  This stores a stack
+         # trace of where it was called, so that later attempts to modify it
+         # can immediately show where it got locked.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $locked{$addr} = "";
+ 
+@@ -6541,7 +6541,7 @@
+     sub carp_if_locked($self) {
+         # Return whether a table is locked or not, and, by the way, complain
+         # if is locked
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return 0 if ! $locked{$addr};
+         Carp::my_carp_bug("Can't modify a locked table. Stack trace of locking:\n$locked{$addr}\n\n");
+@@ -6549,7 +6549,7 @@
+     }
+ 
+     sub set_file_path($self, @path) { # Set the final directory path for this table
+-        @{$file_path{pack 'J', refaddr $self}} = @path;
++        @{$file_path{refaddr $self}} = @path;
+         return
+     }
+ 
+@@ -6691,7 +6691,7 @@
+                                     Write_As_Invlist => 0,
+                                     %args);
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $anomalous_entries{$addr} = [];
+         $default_map{$addr} = $default_map;
+@@ -6751,7 +6751,7 @@
+     sub append_to_body($self) {
+         # Adds to the written HERE document of the table's body any anomalous
+         # entries in the table..
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return "" unless @{$anomalous_entries{$addr}};
+         return join("\n", @{$anomalous_entries{$addr}}) . "\n";
+@@ -6800,7 +6800,7 @@
+                             . " present, must be 'full_name'");
+         }
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Convert the input to the standard equivalent, if any (won't have any
+         # for $STRING properties)
+@@ -6845,7 +6845,7 @@
+ 
+     sub to_output_map($self) {
+         # Returns boolean: should we write this map table?
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # If overridden, use that
+         return $to_output_map{$addr} if defined $to_output_map{$addr};
+@@ -6923,7 +6923,7 @@
+         # No sense generating a comment if aren't going to write it out.
+         return if ! $self->to_output_map;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $property = $self->property;
+ 
+@@ -7102,7 +7102,7 @@
+         # Called in the middle of write when it finds a range it doesn't know
+         # how to handle.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $type = $range->type;
+ 
+@@ -7244,7 +7244,7 @@
+         # be for all ranges missing from it.  It also includes any code points
+         # which have map_types that don't go in the main table.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $name = $self->property->swash_name;
+ 
+@@ -7343,7 +7343,7 @@
+     sub write($self) {
+         # Write the table to the file.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Clear the temporaries
+         undef @multi_code_point_maps;
+@@ -7610,7 +7610,7 @@
+                                       Format => $EMPTY_FORMAT,
+                                       Write_As_Invlist => 1,
+                                       );
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $conflicting{$addr} = [ ];
+         $equivalents{$addr} = [ ];
+@@ -7774,7 +7774,7 @@
+                                         # be an optional parameter.
+         Carp::carp_extra_args(\@_) if main::DEBUG && @_;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Check if the conflicting name is exactly the same as any existing
+         # alias in this table (as long as there is a real object there to
+@@ -7817,7 +7817,7 @@
+         }
+ 
+         # Two tables are equivalent if they have the same leader.
+-        return $leader{pack 'J', refaddr $self} == $leader{pack 'J', refaddr $other};
++        return $leader{refaddr $self} == $leader{refaddr $other};
+         return;
+     }
+ 
+@@ -7854,7 +7854,7 @@
+         my $are_equivalent = $self->is_set_equivalent_to($other);
+         return if ! defined $are_equivalent || $are_equivalent;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+         my $current_leader = ($related) ? $parent{$addr} : $leader{$addr};
+ 
+         if ($related) {
+@@ -7893,8 +7893,8 @@
+             return;
+         }
+ 
+-        my $leader = pack 'J', refaddr $current_leader;
+-        my $other_addr = pack 'J', refaddr $other;
++        my $leader = refaddr $current_leader;
++        my $other_addr = refaddr $other;
+ 
+         # Any tables that are equivalent to or children of this table must now
+         # instead be equivalent to or (children) to the new leader (parent),
+@@ -7911,7 +7911,7 @@
+             next if $table == $other;
+             trace "setting $other to be the leader of $table, status=$status" if main::DEBUG && $to_trace;
+ 
+-            my $table_addr = pack 'J', refaddr $table;
++            my $table_addr = refaddr $table;
+             $leader{$table_addr} = $other;
+             $matches_all{$table_addr} = $matches_all;
+             $self->_set_range_list($other->_range_list);
+@@ -7944,7 +7944,7 @@
+             Carp::my_carp_bug("Can't set $self to be the complement of $other, which itself is the complement of " . $other->complement);
+             return;
+         }
+-        $complement{pack 'J', refaddr $self} = $other;
++        $complement{refaddr $self} = $other;
+ 
+         # Be sure the other property knows we are depending on them; or the
+         # other table if it is one in the current property.
+@@ -8232,7 +8232,7 @@
+ 
+         return unless $debugging_build;
+ 
+-        my $addr = pack 'J', refaddr $leader;
++        my $addr = refaddr $leader;
+ 
+         if ($leader{$addr} != $leader) {
+             Carp::my_carp_bug(<<END
+@@ -8300,7 +8300,7 @@
+                 && $parent == $property->table('N')
+                 && defined (my $yes = $property->table('Y')))
+             {
+-                my $yes_addr = pack 'J', refaddr $yes;
++                my $yes_addr = refaddr $yes;
+                 @yes_perl_synonyms
+                     = grep { $_->property == $perl }
+                                     main::uniques($yes,
+@@ -8316,12 +8316,12 @@
+             my @conflicting;        # Will hold the table conflicts.
+ 
+             # Look at the parent, any yes synonyms, and all the children
+-            my $parent_addr = pack 'J', refaddr $parent;
++            my $parent_addr = refaddr $parent;
+             for my $table ($parent,
+                            @yes_perl_synonyms,
+                            @{$children{$parent_addr}})
+             {
+-                my $table_addr = pack 'J', refaddr $table;
++                my $table_addr = refaddr $table;
+                 my $table_property = $table->property;
+ 
+                 # Tables are separated by a blank line to create a grouping.
+@@ -8786,7 +8786,7 @@
+         my %args = @_;
+ 
+         $self = bless \do { my $anonymous_scalar }, $class;
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         $directory{$addr} = delete $args{'Directory'};
+         $file{$addr} = delete $args{'File'};
+@@ -8849,7 +8849,7 @@
+             return;
+         }
+         else {
+-            $map{pack 'J', refaddr $self}->delete_range($other, $other);
++            $map{refaddr $self}->delete_range($other, $other);
+         }
+         return $self;
+     }
+@@ -8862,7 +8862,7 @@
+         my $name = shift;
+         my %args = @_;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $table = $table_ref{$addr}{$name};
+         my $standard_name = main::standardize($name);
+@@ -8936,7 +8936,7 @@
+ 
+     sub delete_match_table($self, $table_to_remove) {
+         # Delete the table referred to by $2 from the property $1.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Remove all names that refer to it.
+         foreach my $key (keys %{$table_ref{$addr}}) {
+@@ -8951,7 +8951,7 @@
+     sub table($self, $name) {
+         # Return a pointer to the match table (with name given by the
+         # parameter) associated with this property; undef if none.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         return $table_ref{$addr}{$name} if defined $table_ref{$addr}{$name};
+ 
+@@ -8969,7 +8969,7 @@
+         # Return a list of pointers to all the match tables attached to this
+         # property
+ 
+-        return main::uniques(values %{$table_ref{pack 'J', refaddr shift}});
++        return main::uniques(values %{$table_ref{refaddr shift}});
+     }
+ 
+     sub directory {
+@@ -8978,7 +8978,7 @@
+         # priority;  'undef' is returned if the type isn't defined;
+         # or $map_directory for everything else.
+ 
+-        my $addr = pack 'J', refaddr shift;
++        my $addr = refaddr shift;
+ 
+         return $directory{$addr} if defined $directory{$addr};
+         return undef if $type{$addr} == $UNKNOWN;
+@@ -8995,7 +8995,7 @@
+         # but otherwise the standard name is used.  This is different from the
+         # external_name, so that the rest of the files, like in lib can use
+         # the standard name always, without regard to historical precedent.
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Swash names are used only on either
+         # 1) regular or internal-only map tables
+@@ -9016,7 +9016,7 @@
+         # The whole point of this pseudo property is match tables.
+         return 1 if $self == $perl;
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # Don't generate tables of code points that match the property values
+         # of a string property.  Such a list would most likely have many
+@@ -9043,7 +9043,7 @@
+             return;
+         }
+ 
+-        return $map{pack 'J', refaddr $self}->map_add_or_replace_non_nulls($map{pack 'J', refaddr $other});
++        return $map{refaddr $self}->map_add_or_replace_non_nulls($map{refaddr $other});
+     }
+ 
+     sub set_proxy_for {
+@@ -9079,7 +9079,7 @@
+             return;
+         }
+ 
+-        $type{pack 'J', refaddr $self} = $type;
++        $type{refaddr $self} = $type;
+         return if $type != $BINARY && $type != $FORCED_BINARY;
+ 
+         my $yes = $self->table('Y');
+@@ -9118,7 +9118,7 @@
+         my $map = shift;    # What the range maps to.
+         # Rest of parameters passed on.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         # If haven't the type of the property, gather information to figure it
+         # out.
+@@ -9167,7 +9167,7 @@
+         # We have been keeping track of what the property values have been,
+         # and now have the necessary information to figure out the type.
+ 
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+ 
+         my $type = $type{$addr};
+ 
+@@ -9223,7 +9223,7 @@
+ 
+     # $reaons - Ignored unless suppressing
+     sub set_fate($self, $fate, $reason=undef) {
+-        my $addr = pack 'J', refaddr $self;
++        my $addr = refaddr $self;
+         if ($fate >= $SUPPRESSED) {
+             $why_suppressed{$self->complete_name} = $reason;
+         }
+@@ -9301,7 +9301,7 @@
+         *$sub = sub {
+             use strict "refs";
+             my $self = shift;
+-            return $map{pack 'J', refaddr $self}->$sub(@_);
++            return $map{refaddr $self}->$sub(@_);
+         }
+     }
+ 
+@@ -9714,7 +9714,7 @@
+         else {
+ 
+             # Keep track of cycles in the input, and refuse to infinitely loop
+-            my $addr = pack 'J', refaddr $item;
++            my $addr = refaddr $item;
+             if (defined $main::already_output{$addr}) {
+                 return "${indent}ALREADY OUTPUT: $item\n";
+             }
+@@ -9832,7 +9832,7 @@
+     # Dump inside-out hashes in an object's state by converting them to a
+     # regular hash and then calling simple_dumper on that.
+ 
+-    my $addr = pack 'J', refaddr $object;
++    my $addr = refaddr $object;
+ 
+     my %hash;
+     foreach my $key (keys %$fields_ref) {
+@@ -9853,7 +9853,7 @@
+         }
+         else {
+             my $ref = ref $$which;
+-            my $addr = pack 'J', refaddr $$which;
++            my $addr = refaddr $$which;
+             $$which = "$ref ($addr)";
+         }
+     }
+@@ -10857,7 +10857,7 @@
+                     $file->carp_bad_line("Unexpected property '$property_name'.  Skipped");
+                     next LINE;
+                 }
+-                $property_addr = pack 'J', refaddr $property_object;
++                $property_addr = refaddr $property_object;
+ 
+                 # Defer changing names until have a line that is acceptable
+                 # (the 'next' statement above means is unacceptable)
+@@ -10909,7 +10909,7 @@
+                                             if $file->has_missings_defaults;
+                     foreach my $default_ref (@missings_list) {
+                         my $default = $default_ref->[0];
+-                        my $addr = pack 'J', refaddr property_ref($default_ref->[1]);
++                        my $addr = refaddr property_ref($default_ref->[1]);
+ 
+                         # For string properties, the default is just what the
+                         # file says, but non-string properties should already
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -60,0 +60,1 @@
+local/mktables-pack-J.diff
